app/scripts/modules/core/src/header/SpinnakerHeader.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+import { UISref, UISrefActive } from '@uirouter/react';
+import { UIRouterContext } from '@uirouter/react-hybrid';
+
+import { NgReact } from 'core/reactShims';
+
+import './SpinnakerHeader.css';
+
+export interface ISpinnakerHeaderState {
+  navExpanded: boolean;
+}
+
+@UIRouterContext
+@autoBindMethods
+export class SpinnakerHeader extends React.Component<{}, ISpinnakerHeaderState> {
+
+  constructor() {
+    super();
+    this.state = {
+      navExpanded: !this.isDevicePhoneOrSmaller()
+    };
+  }
+
+  public isDevicePhoneOrSmaller(): Boolean {
+    const bodyStyles = window.getComputedStyle(document.body);
+    const isPhone = bodyStyles.getPropertyValue('--is-phone');
+    return isPhone.toLowerCase() === 'true';
+  }
+
+  public toggleNavItems(): void {
+    this.setState({
+      navExpanded: !this.state.navExpanded
+    });
+  }
+
+  public render(): React.ReactElement<SpinnakerHeader> {
+    const { UserMenu, GlobalSearch, WhatsNew } = NgReact;
+
+    return (
+      <nav
+        className="container styleguide SpinnakerHeader"
+        role="navigation"
+        aria-label="Main Menu"
+      >
+        <div className="navbar-header horizontal middle">
+          <a className="navbar-brand flex-1" href="#">SPINNAKER</a>
+          <button type="button" className="navbar-toggle" onClick={this.toggleNavItems}>
+            <span className="icon-bar" />
+            <span className="icon-bar" />
+            <span className="icon-bar" />
```

Comment:
think of all the savings from using `<i className="fa fa-bars"/>`

just think about it, you don't have to do anything about it

Reply:
Needs its own diff. More styling changes required to get the icon working.